nilearn/surface/tests/test_surface.py

```diff
@@ -76,7 +76,7 @@ def test_load_surf_data_array():
 
 def test_load_surf_data_file_nii_gii():
     # test loading of fake data from gifti file
-    filename_gii = tempfile.mktemp(suffix='.gii')
+    fd_gii, filename_gii = tempfile.mkstemp(suffix='.gii')
```

Comment:
in all the uses of `mkstemp` below I would suggest closing the file right after creating it.

also, this and most of the nilearn tests we see here leak resources: if the test fails the temporary files are not removed.

it would be better to create these files in pytest's `tmp_path` and do not bother to remove them (it is not removed by default so that it can be inspected when investigating failures, but at least all temp files are in one place in the `/tmp/pytest-of-$USER` and it is easy to remove it or configure it to be removed automatically)


an alternative would be rely on a context manager to remove them (not `tempfile.NamedTemporaryFile`, because it can either remove the file as soon as it is closed or never remove it), but I prefer the pytest tmp_path solution.

I guess it should be done in a separate pr to avoid slowing down this one; if you agree I'll open an issue